wcfsetup/install/files/lib/system/flood/FloodControl.class.php

```diff
@@ -0,0 +1,241 @@
+<?php
+namespace wcf\system\flood;
+use wcf\data\object\type\ObjectTypeCache;
+use wcf\system\SingletonFactory;
+use wcf\system\WCF;
+use wcf\util\DateUtil;
+use wcf\util\UserUtil;
+
+/**
+ * Flood control tracks the times contents are created by users.
+ * 
+ * @author      Matthias Schmidt
+ * @copyright   2001-2020 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package     WoltLabSuite\Core\System\Flood
+ * @since       5.4
+ */
+class FloodControl extends SingletonFactory {
+	const PRUNE_TIME = TIME_NOW - 31 * 86400;
+	
+	/**
+	 * Returns the number of contents by a certain identifier type within a certain
+	 * period of time `[$time-$interval, $time]` and the earliest time within the period content
+	 * was created.
+	 */
+	protected function countContentByIdentifier(string $objectType, string $identifier, \DateInterval $interval, int $time): array {
```

Comment:
```suggestion
	protected function countContentByIdentifier(string $objectType, string $identifier, \DateInterval $interval, \DateTime $time): array {
```

More strongly typed, especially since you call `getDateTimeByTimestamp` anyway?

Reply:
With the current implementation, the `\DateTime` object is only created once in this method but with this change, each calling methods would have to change the timestamp to a `\DateTime` object.